packages/common/src/models/USDCTransactions.ts

```diff
@@ -0,0 +1,79 @@
+import { Nullable } from '../utils/typeUtils'
+
+import { StringUSDC } from './Wallet'
+
+export enum USDCTransactionType {
+  PURCHASE = 'PURCHASE',
+  PURCHASE_CONTENT = 'PURCHASE_CONTENT',
+  TRANSFER = 'TRANSFER'
+}
+
+export enum USDCTransactionMethod {
+  // Transfer methods
+  SEND = 'SENT',
+  RECEIVE = 'RECEIVED',
+
+  // Purchase Methods
+  STRIPE = 'STRIPE'
+}
+
+export enum USDCTransactionMetadataType {
+  PURCHASE_USDC = 'PURCHASE_USDC',
+  PURCHASE_CONTENT = 'PURCHASE_CONTENT',
+  TRANSER = 'TRANSFER'
+}
+
+export type InAppUSDCPurchaseMetadata = {
+  discriminator: USDCTransactionMetadataType.PURCHASE_USDC
+  usd: string
+  usdc: StringUSDC
+  purchaseTransactionId: string
+}
+
+export enum USDCContentPurchaseType {
+  TRACK = 'TRACK'
+}
+
+export type USDCPurchaseContentMetadata = {
+  discriminator: USDCTransactionMetadataType.PURCHASE_CONTENT
+  amount: StringUSDC
+  senderUserId: string
+  receiverUserId: string
+  purchaseType: USDCContentPurchaseType
+  contentId: string
+}
+
+export type USDCTransferMetadata = {
+  discriminator: USDCTransactionMetadataType.TRANSER
+  amount: StringUSDC
+  destination: string
+}
+
+export type USDCTransactionDetails =
+  | {
+      signature: string
+      transactionType: USDCTransactionType.PURCHASE
+      method: USDCTransactionMethod.STRIPE
+      date: string
+      change: StringUSDC
+      balance: StringUSDC
+      metadata?: Nullable<InAppUSDCPurchaseMetadata>
+    }
+  | {
+      signature: string
+      transactionType: USDCTransactionType.PURCHASE_CONTENT
+      method: USDCTransactionMethod.SEND | USDCTransactionMethod.RECEIVE
+      date: string
+      change: StringUSDC
+      balance: StringUSDC
+      metadata?: Nullable<USDCPurchaseContentMetadata>
+    }
+  | {
+      signature: string
+      transactionType: USDCTransactionType.TRANSFER
+      method: USDCTransactionMethod.SEND | USDCTransactionMethod.RECEIVE
```

Comment:
Don't think we will support receiving USDC (since we don't expose your USDC wallet anywhere), so should be able to cut this `RECEIVE`

Reply:
This pattern came from audio transactions, and I was initially confused by it. It looks like we write two transactions each time Audio is sent (tipping), one for the sender and one for the receiver. I think having the `receive` item helps simplify the transactions list for any individual user because we select by `user_bank`, and this would correctly show the times you received USDC via a purchase.